src/js/view/controls/components/menu/menu.js

```diff
@@ -0,0 +1,499 @@
+                    activeItem.deactivate();
+                });
+                menuItem.activate();
+                this.mainMenu.close(evt);
+            });
+
+            menuItem.ui.on('keydown', (event) => {
+                const focusElement = (ele, i) => {
+                    if (ele) {
+                        ele.focus();
+                    } else if (i !== undefined) {
+                        menuItems[i].el.focus();
+                    }
+                };
+                const evt = event.sourceEvent;
+                const nextItem = nextSibling(categoryButtonElement);
+                const prevItem = previousSibling(categoryButtonElement);
+                const nextSubItem = nextSibling(evt.target);
+                const prevSubItem = previousSibling(evt.target);
+                const key = evt.key.replace(/(Arrow|ape)/, '');
+        
+                switch (key) {
+                    case 'Tab':
+                        focusElement(evt.shiftKey ? prevItem : nextItem);
+                        break;
+                    case 'Left':
+                        focusElement(prevItem || 
+                            previousSibling(document.getElementsByClassName('jw-icon-settings')[0]));
+                        break;
+                    case 'Up':
+                        focusElement(prevSubItem, menuItems.length - 1);
+                        break;
+                    case 'Right':
+                        focusElement(nextItem);
+                        break;
+                    case 'Down':
+                        focusElement(nextSubItem, 0);
+                        break;
+                    default:
+                        break;
+                }
+                evt.preventDefault();
+                if (key !== 'Esc') {
+                    // only bubble event if esc key was pressed
+                    evt.stopPropagation();
+                }
+                
+            });
+
+            return menuItem;
+        });
+
+        return menuItems;
+    }
+    defaultChild() {
+        return this.children.quality || this.children.captions || this.children.audioTracks || this.children.sharing;
+    }
+    setMenuItems(menuItems, initialSelectionIndex) {
+        if (!menuItems) {
+            this.removeMenu();
+        } else {
+            this.items = [];
+            emptyElement(this.el);
+            menuItems.forEach(menuItem => {
+                this.items.push(menuItem);
+                this.el.appendChild(menuItem.el);
+            });
+            if (initialSelectionIndex > -1) {
+                menuItems[initialSelectionIndex].activate();
+            }
+            this.categoryButton.show();
+        }
+    }
+    appendMenu(childMenu) {
+        if (!childMenu) {
+            return;
+        }
+        // eslint-disable-next-line no-shadow
+        const { el, name, categoryButton } = childMenu;
+        this.children[name] = childMenu;
+        if (categoryButton) {
```

Comment:
can i get a rundown on what this logic is doing? i'm a lil confused

Reply:
bump on this question!

Reply:
If the menu has a category button, append it to the main menu topbar with respect to our previous ordering logic. (Quality first, else put it before the close button)